metaswitch/common/utils.py

```diff
@@ -399,3 +400,22 @@ def sigusr1_handler(sig, stack):
         stack_dump = "Caught SIGUSR1\n" + "".join(traceback.format_stack(stack))
         write_core_file(process_name, stack_dump)
     signal.signal(signal.SIGUSR1, sigusr1_handler)
+
+def write_pid_file(filename):
```

Comment:
I think this function name should mention that we're also grabbing a lock. `acquire_pid_file_lock`? I think this also makes it clear that the function returns a RAII guard that the caller must take care of (though I think you should add a comment about that to the docstring as well).